packages/run-protocol/src/vaultFactory/liquidateIncrementally.js

```diff
@@ -0,0 +1,322 @@
+        BP2,
+      ),
+    );
+
+  /**
+   *
+   * @param {ZCFSeat} seat
+   * @param {Amount} originalDebt
+   * @yields {LiquidationStep}
+   */
+  async function* processTranches(seat, originalDebt) {
+    let proceedsSoFar = seat.getAmountAllocated('Out');
+    let toSell = seat.getAmountAllocated('In');
+    while (
+      !AmountMath.isGTE(proceedsSoFar, originalDebt) &&
+      !AmountMath.isEmpty(toSell)
+    ) {
+      const debt = AmountMath.subtract(originalDebt, proceedsSoFar);
+      const { Secondary: poolCollateral, Central: poolCentral } = await E(
+        amm,
+      ).getPoolAllocation(toSell.brand);
+      const maxAllowedTranche = maxTrancheWithFees(poolCollateral);
+      trace('TRANCHE', asFloat(maxAllowedTranche.value, 100000n));
+      const tranche = AmountMath.min(maxAllowedTranche, toSell);
+      // compute the expected proceeds from the AMM for tranche
+      const minAmmProceeds = computeAMMProceeds(
+        poolCentral,
+        poolCollateral,
+        tranche,
+        debt,
+      );
+      // true if the minimum proceeds will pay off the debt
+      const isLast = AmountMath.isGTE(minAmmProceeds, debt);
+
+      // this could use a unit rather than the tranche size to run concurrently
+      /** @type PriceQuote */
+      const oracleQuote = await E(priceAuthority).quoteGiven(
+        tranche,
+        debtBrand,
+      );
+      // TODO Make a governance parameter
+      const oracleLimit = ceilMultiplyBy(
+        getAmountOut(oracleQuote),
+        ORACLE_LIMIT_BP,
+      );
+
+      trace('TRANCHE', {
+        debt,
+        tranche,
+        minAmmProceeds,
+        oracleLimit,
+        isLast,
+        poolCentral,
+        poolCollateral,
+      });
+      if (AmountMath.isGTE(minAmmProceeds, oracleLimit)) {
+        yield {
+          collateral: tranche,
+          ammProceeds: minAmmProceeds,
+          debt,
+          oracleLimit,
+          isLast,
+        };
+      } else {
+        trace('SKIP');
+      }
+      proceedsSoFar = seat.getAmountAllocated('Out');
+      toSell = seat.getAmountAllocated('In');
+      if (
+        AmountMath.isGTE(proceedsSoFar, originalDebt) ||
+        AmountMath.isEmpty(toSell)
+      ) {
+        trace('exiting async loop');
+        return;
+      }
+      await nextBlock();
+    }
+  }
+
+  async function sellTranche(debtorSeat, collateral, debt, oracleLimit) {
+    const swapInvitation = E(amm).makeSwapInvitation();
+    const want = AmountMath.min(debt, oracleLimit);
+    // TODO if the debt shouldn't require all the collatearal to cover it,
+    // this may not provide relevant offer safety
+    const proposal = {
+      give: { In: collateral },
+      want: { Out: want },
+    };
+    const { deposited, userSeatPromise: liqSeat } = await offerTo(
+      zcf,
+      swapInvitation,
+      undefined, // The keywords were mapped already
+      proposal,
+      debtorSeat,
+      debtorSeat,
+      { stopAfter: debt },
```

Comment:
The fact that adding a parameter made you fill in a previously-optional parameter argues that `offerTo` should take a bag of parameters.   Certainly not something to address here.